### PR TITLE
fix helm install failure 

### PR DESCRIPTION
**Pull request by jiaqiluo**
issue: https://github.com/rancher/rancher/issues/33831

Problem:
The current design is to skip the creation of all CRDs if the API group version `monitoring.coreos.com/v1` exists in the cluster.

If one of the crd, for example, the crd `servicemonitors.monitoring.coreos.com/v1` does not exist, Helm install will fail and report the following error:

```
Failed to install app cluster-alerting. Error: validation failed: unable to recognize "": no matches for kind "ServiceMonitor" in the API group version "monitoring.coreos.com/v1"
```

Fix:

Instead of checking the API group, we now check each individual kind and create the missing one while installing the app.  ( **see the 3rd commit for the fix**)

### CODE REVIEW

File: charts/rancher-monitoring/v0.3.1/Chart.yaml [hunk 1]
```diff
@@ -7,8 +7,8 @@ maintainers:
 name: rancher-monitoring
 sources:
 - https://github.com/coreos/prometheus-operator
-version: 0.3.0
-appVersion: 0.3.0
+version: 0.3.1
```
Comment (aiyengar2) [inline charts/rancher-monitoring/v0.3.1/Chart.yaml:10]: Need to update question.yml as well

**Review by aiyengar2 — CHANGES_REQUESTED** [review]

File: charts/rancher-monitoring/v0.3.1/Chart.yaml (hunk 1 above)
Comment (jiaqiluo) [reply charts/rancher-monitoring/v0.3.1/Chart.yaml:10]: fixed!

**Review by brendarearden — DISMISSED** [review]
LGTM

**Review by aiyengar2 — DISMISSED** [review]
LGTM

File: charts/rancher-monitoring/v0.3.1/templates/crds.yaml [hunk 2]
```diff
@@ -18,7 +18,9 @@ spec:
     plural: alertmanagers
   scope: Namespaced
   version: v1
+{{- end }}
 ---
```
Comment (cbron) [inline charts/rancher-monitoring/v0.3.1/templates/crds.yaml:21]: These should be included in the if blocks, or you'll get: 
```
---
---
---
```

File: charts/rancher-monitoring/v0.3.1/questions.yml [hunk 3]
```diff
@@ -0,0 +1 @@
+rancher_min_version: 2.6.1
```
Comment (cbron) [inline charts/rancher-monitoring/v0.3.1/questions.yml]: The 0.3.0 chart should have a max of 2.6.0

File: charts/rancher-monitoring/v0.3.1/templates/crds.yaml (hunk 2 above)
Comment (jiaqiluo) [reply charts/rancher-monitoring/v0.3.1/templates/crds.yaml:21]: fixed.

File: charts/rancher-monitoring/v0.3.1/questions.yml (hunk 3 above)
Comment (jiaqiluo) [reply charts/rancher-monitoring/v0.3.1/questions.yml]: `rancher_max_version: 2.6.0` is added to monitoring 0.3.0

**Review by cbron — APPROVED** [review]

**Pull request merged**